### PR TITLE
feat(dropdown-item): add disabled support

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #6667

## Summary

This allows disabling dropdown items.

### CODE REVIEW

File: packages/calcite-components/src/components/dropdown/dropdown.tsx [hunk 1]
```diff
@@ -374,27 +374,32 @@ export class Dropdown
     this.closeCalciteDropdown();
   }
 
+  private getTraversableItems(): HTMLCalciteDropdownItemElement[] {
+    return this.items.filter((item) => !item.disabled);
```
Comment (driskull) [inline packages/calcite-components/src/components/dropdown/dropdown.tsx:378]: could also be beneficial to check for any `item.hidden` items. I could see this as some sort of dom utility potentially.

**Review by driskull — APPROVED** [review]
👍

File: packages/calcite-components/src/components/dropdown/dropdown.tsx (hunk 1 above)
Comment (jcfranco) [reply packages/calcite-components/src/components/dropdown/dropdown.tsx:378]: Agreed. I'll add the `hidden` check here and will look into extracting a util in a separate PR. I think a few keyboard navigable components could benefit from this. 🎉

File: packages/calcite-components/src/components/dropdown-item/dropdown-item.e2e.ts [hunk 2]
```diff
@@ -14,6 +14,10 @@ describe("calcite-dropdown-item", () => {
     focusable(`calcite-dropdown-item`);
   });
 
+  describe("disabled", () => {
+    disabled(`calcite-dropdown-item`);
```
[the remark this answers is not in the document]
Comment (jcfranco) [reply packages/calcite-components/src/components/dropdown-item/dropdown-item.e2e.ts:18]: I'll do this in a separate PR. I'd like to have this automated if possible.

**Pull request merged**